pyro/ops/tensor_utils.py

```diff
@@ -13,3 +13,30 @@ def block_diag(m):
     B, M, N = m.shape
     eye = torch.eye(B, dtype=m.dtype, device=m.device).reshape(B, 1, B, 1)
     return (m.unsqueeze(-2) * eye).reshape(B * M, B * N)
+
+
+def _complex_mul(a, b):
+    ar, ai = a.unbind(-1)
+    br, bi = b.unbind(-1)
+    # FIXME should b be conjugated?
+    # bi = -bi  # Conjugates b.
+    return torch.stack([ar * br - ai * bi, ar * bi + ai * br], dim=-1)
+
+
+def conv1d_fft(signal, kernel):
+    """
+    Computes the 1-d convolution of signal by kernel using FFTs.
+    The two arguments should have the same rightmost dim, but may otherwise be
+    arbitrarily broadcastable.
+
+    :param torch.Tensor signal: A signal to convolve.
+    :param torch.Tensor kernel: A convolution kernel.
+    :return: A tensor of shape
+        ``broadcast_shape(convolve, kernel.shape)``
+    :rtype torch.Tensor:
+    """
+    assert signal.size(-1) == kernel.size(-1)
+    f_signal = torch.rfft(torch.cat([signal, torch.zeros_like(signal)], dim=-1), 1)
+    f_kernel = torch.rfft(torch.cat([kernel, torch.zeros_like(kernel)], dim=-1), 1)
+    f_result = _complex_mul(f_signal, f_kernel)
+    return torch.irfft(f_result, 1)[..., :signal.size(-1)]
```

Comment:
I guess you need the output to have the same last dimension as the signal. Probably you will want to take the center of the f_result?

Reply:
The updated truncation logic now matches scipy.signal.convolve in size.